src/code/z_skin_awb.c

```diff
@@ -1,13 +1,204 @@
 #include "global.h"
+#include "z64skin.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_awb/func_80138410.s")
+void func_80138410(Skin* skin) {
+    skin->skeletonHeader = NULL;
+    skin->limbCount = 0;
+    skin->vtxTable = NULL;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_awb/func_80138424.s")
+/**
+ * Initialises the Vtx buffers used for limb at index `limbIndex`
+ */
+void Skin_InitAnimatedLimb(GameState* gameState, Skin* skin, s32 limbIndex) {
+    s32 i;
+    SkinLimb** skeleton = Lib_SegmentedToVirtual(skin->skeletonHeader->segment);
+    SkinAnimatedLimbData* animatedLimbData =
+        Lib_SegmentedToVirtual(((SkinLimb*)Lib_SegmentedToVirtual(skeleton[limbIndex]))->segment);
+    SkinLimbModif* limbModifications = Lib_SegmentedToVirtual(animatedLimbData->limbModifications);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_awb/func_8013859C.s")
+    for (i = 0; i < ARRAY_COUNT(skin->vtxTable->buf); i++) {
+        Vtx* vertices = skin->vtxTable[limbIndex].buf[i];
+        SkinLimbModif* modifEntry;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_awb/func_80138700.s")
+        for (modifEntry = limbModifications; modifEntry < &limbModifications[animatedLimbData->limbModifCount];
+             modifEntry++) {
+            SkinVertex* skinVertices = Lib_SegmentedToVirtual(modifEntry->skinVertices);
+            SkinVertex* skinVtxEntry;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_awb/func_801387D4.s")
+            for (skinVtxEntry = skinVertices; skinVtxEntry < &skinVertices[modifEntry->vtxCount];) {
+                Vtx* vtx = &vertices[skinVtxEntry->index];
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_awb/func_801388E4.s")
+                skinVtxEntry++;
+
+                vtx->n.flag = 0;
+                vtx->n.tc[0] = skinVtxEntry[-1].s;
+                vtx->n.tc[1] = skinVtxEntry[-1].t;
+                vtx->n.a = skinVtxEntry[-1].alpha;
+            }
+        }
+    }
+}
+
+/**
+ * Initialises a skin skeleton to looping animation, dynamically allocating the frame tables,
+ * and dynamically allocating and initializing the Vtx and SkinLimbVtx buffers for its animated limbs
+ */
```

Comment:
```suggestion
/**
 * Initializes a skin skeleton to looping animation, dynamically allocating the frame tables,
 * and dynamically allocating and initialising the Vtx and SkinLimbVtx buffers for its animated limbs
 */
```
`Initialise` with an `s` and a `z` in the same sentence. Nice. I don't have a preference, as long as its consistent